administrator/components/com_fields/models/fields/type.php

```diff
@@ -91,4 +107,31 @@ protected function getOptions ()
 
 		return $options;
 	}
+
+	/**
+	 * Parses the file with the given path. If it is a class starting with the
+	 * name JFormField and implementing JFormDomfieldinterface, then the class name is returned.
+	 *
+	 * @param string $path
+	 * @return string|boolean
+	 */
+	private function getClassNameFromFile($path)
+	{
+		$tokens = token_get_all(JFile::read($path));
```

Comment:
That's a good point. We should check it's an instance of `JFormField` and `JFormDomfieldinterface`